src/libraries/System.Private.CoreLib/src/System/UInt128.cs

```diff
@@ -1091,6 +1091,11 @@ public static UInt128 Log2(UInt128 value)
         /// <inheritdoc cref="IDivisionOperators{TSelf, TOther, TResult}.op_Division(TSelf, TOther)" />
         public static UInt128 operator /(UInt128 left, UInt128 right)
         {
+            if ((right._lower == 0) && (right._upper == 0))
+            {
+                throw new DivideByZeroException();
+            }
+
             if ((right._upper == 0) && (left._upper == 0))
```

Comment:
Is it better or worse to combine the check of right._upper == 0 with the one above?

Reply:
If we don't combine them, I guess putting the DevideByZero check after the both-int64 check could bring better performance.

Reply:
Checks combined.
However, as a general rule, I prefer to keep input validation first, before routing the call to the `UInt64` divide operator.
@danmoseley Is the check order important here?

Reply:
Area owners should answer that.

Reply:
We should prefer throwing exceptions for "invalid" inputs first.